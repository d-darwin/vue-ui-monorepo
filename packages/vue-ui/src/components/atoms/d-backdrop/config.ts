import { CssPosition } from "./types";
import styles from "./index.css?module";

export default {
  name: "DBackdrop",
  class: styles.dBackdrop,
  opacity: 0.5,
  zIndex: 1000,
  position: CssPosition.fixed,
} as const;
