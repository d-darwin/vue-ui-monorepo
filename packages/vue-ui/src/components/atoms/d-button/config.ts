export default {
  name: "DButton",
  className: "dButton",
};
