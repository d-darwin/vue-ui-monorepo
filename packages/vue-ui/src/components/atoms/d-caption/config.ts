import styles from "./index.css?module";

export default {
  name: "DCaption",
  class: styles["dCaption"],
} as const;
