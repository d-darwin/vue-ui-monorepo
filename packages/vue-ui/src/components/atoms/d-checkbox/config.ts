export default {
  name: "DCheckbox",
  className: "dCheckbox",
  inputClassName: "input",
  labelClassName: "label",
  labelInnerClassName: "labelInner",
  errorClassName: "error",
  iconContainerClassName: "iconContainer",
  iconContainerBackdropClassName: "iconContainerBackdrop",
  iconClassName: "icon",
  checkMark: "âœ“",
};
