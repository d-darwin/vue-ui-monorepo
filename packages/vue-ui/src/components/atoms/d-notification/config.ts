export default {
  name: "DNotification",
  className: "dNotification",
} as const;
