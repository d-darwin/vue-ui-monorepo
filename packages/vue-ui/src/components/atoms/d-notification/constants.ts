export const TYPE = {
  SUCCESS: "success",
  INFO: "info",
  WARNING: "warning",
  ERROR: "error",
} as const;
