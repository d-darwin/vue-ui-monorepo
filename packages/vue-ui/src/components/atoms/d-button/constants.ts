import type { DLoaderProps } from "@darwin-studio/vue-ui/src/components/atoms/d-loader/types";
import config from "./config";
import styles from "./index.css?module";

export const LOADER_DEFAULTS: DLoaderProps = {
  class: [styles[config.loaderClassName]],
};
