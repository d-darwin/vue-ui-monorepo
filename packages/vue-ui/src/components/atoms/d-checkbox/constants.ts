import { DCaptionProps } from "@darwin-studio/vue-ui/src/components/atoms/d-caption/types";
import { TYPE } from "@darwin-studio/vue-ui/src/components/atoms/d-caption/constant";
import config from "@darwin-studio/vue-ui/src/components/atoms/d-switch/config";
import styles from "./index.css?module";

export const DEFAULT_VALUE = "on";

export const CAPTION_DEFAULTS: DCaptionProps = {
  type: TYPE.DANGER,
  class: styles[config.captionClassName],
};
