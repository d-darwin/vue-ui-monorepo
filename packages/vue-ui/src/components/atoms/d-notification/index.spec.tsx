import { shallowMount } from "@vue/test-utils";
// TODO: get @darwin-studio/ui-codegen paths from config.json
import { FONT } from "@darwin-studio/ui-codegen/dist/constants/font"; // TODO: shorter path, default export ???
import prepareCssClassName from "@darwin-studio/ui-codegen/src/utils/prepareCssClassName"; // TODO: shorter path ???
import DNotification from "@/components/atoms/d-notification";
import codegenConfig from "@darwin-studio/ui-codegen/config.json";
import {
  baseClassCase,
  propContentCase,
  contentHtmlCase,
  slotDefaultCase,
  tagCase,
} from "@/utils/test-case-factories";
import config from "./config";

describe("DNotification", () => {
  const wrapper = shallowMount(DNotification);

  baseClassCase(wrapper, config.className);

  propContentCase(wrapper);

  contentHtmlCase(wrapper);

  slotDefaultCase(DNotification);

  // TODO: classes are not rendered in jest ???
  it("Renders props.font to font class when passed", async () => {
    const font = FONT.HUGE;
    await wrapper.setProps({ font }); // TODO: why it doesn't work with composition api ???
    const className = prepareCssClassName(
      codegenConfig.TOKENS.FONT.CSS_CLASS_PREFIX,
      font
    );
    expect(wrapper.classes()).toContain(className);
  });

  tagCase(wrapper);
});
