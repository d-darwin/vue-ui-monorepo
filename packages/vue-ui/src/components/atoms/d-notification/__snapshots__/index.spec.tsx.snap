// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DNotification Should append the notification to the props.target 1`] = `
<div id="custom-target"><transition-stub enteractiveclass="transitionEnterActive" leaveactiveclass="transitionLeaveActive" appear="false" persisted="false" css="true">
    <div role="alert" class="dNotification top-right info colorSchemePrimary fontMedium paddingDefault paddingDefaultMedium roundingSmall sizeMedium transitionFast"></div>
  </transition-stub></div>
<div data-v-app=""><!--teleport start--><!--teleport end--></div>
`;
